Containerfile

```diff
@@ -1,19 +1,14 @@
 # SPDX-License-Identifier: Apache-2.0
-
-FROM quay.io/projectquay/golang:1.22 AS makefile
```

Comment:
This was used by containerized-build.sh, which is being deleted by this PR. Why is it being deleted?

Reply:
I was confused why that file was needed at all, it's just a podman build command. It seems like the same could be achieved by just running a make command if we really want to simplify it:
```
.PHONY: container-build
container-build:
    podman build -t kubesan:latest .
```

Reply:
containerized-build.sh was added recently because the CI environment does not have the right golang version available. It's the entry point that the Jenkins CI calls.

If things can be simplified by just having the CI call `podman build -t kubesan:latest .` directly, then that would be nice. But I suspect the CI will break if we merge this PR without CI changes.

Adding @fabbione and Chandan Somani to see what they think.

Reply:
Historical background: before containerized-build.sh was added, 'make build' used to kick off podman unconditionally, but only after also running go to build controller-gen locally.  That mix (the developer's own go building one part of the tools, then the golang in the container running the rest of the build) is not ideal.  containerized-build.sh made it so that we could guarantee that the same version of Go was used for everything in the build.

it would not hurt to have 'make container-build' for developers to use (although the Containerfile must not use it; that would be a recursive nightmare).  Meanwhile, in a CI, you never want to start with bare 'make', but only things that will use the container (but within the container, calling bare 'make' is okay).

Reply:
Mentioned in another comment too. CI is here to support you and not to make things more complex. building in CI is a one liner change. Once you decide the approach, I will gladly patch the pipeline.